### ISSUE TITLE
[🐛] 🔥 Production crash on iOS firebasePerformance NSKeyValuePopPendingNotificationLocal

### ISSUE THREAD

**Issue by wilau2** [invertase/react-native-firebase#4458]
## Issue

Monitoring production crash: 
![image](https://user-images.githubusercontent.com/5473183/97207203-1d587d00-1790-11eb-96b9-ce13c9d6cd45.png)

Seems to be related to:
https://github.com/firebase/firebase-ios-sdk/issues/6734

---

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
 	"@react-native-firebase/app": "8.4.5",
    "@react-native-firebase/analytics": "7.6.7",
    "@react-native-firebase/crashlytics": "8.4.9",
    "@react-native-firebase/messaging": "7.8.11",
    "@react-native-firebase/perf": "7.4.8",
```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
  - Firebase/Analytics (6.30.0):
    - Firebase/Core
  - Firebase/Core (6.30.0):
    - Firebase/CoreOnly
    - FirebaseAnalytics (= 6.7.2)
  - Firebase/CoreOnly (6.30.0):
    - FirebaseCore (= 6.10.0)
  - Firebase/Crashlytics (6.30.0):
    - Firebase/CoreOnly
    - FirebaseCrashlytics (~> 4.4.0)
  - Firebase/Messaging (6.30.0):
    - Firebase/CoreOnly
    - FirebaseMessaging (~> 4.6.1)
  - Firebase/Performance (6.30.0):
    - Firebase/CoreOnly
    - FirebasePerformance (~> 3.3.0)
  - FirebaseABTesting (4.2.0):
    - FirebaseCore (~> 6.10)
  - FirebaseAnalytics (6.7.2):
    - FirebaseCore (~> 6.8)
    - FirebaseInstallations (~> 1.4)
    - GoogleAppMeasurement (= 6.7.2)
    - GoogleUtilities/AppDelegateSwizzler (~> 6.7)
    - GoogleUtilities/MethodSwizzler (~> 6.7)
    - GoogleUtilities/Network (~> 6.7)
    - "GoogleUtilities/NSData+zlib (~> 6.7)"
    - nanopb (~> 1.30905.0)
  - FirebaseCore (6.10.0):
    - FirebaseCoreDiagnostics (~> 1.3)
    - GoogleUtilities/Environment (~> 6.7)
    - GoogleUtilities/Logger (~> 6.7)
  - FirebaseCoreDiagnostics (1.5.0):
    - GoogleDataTransport (~> 7.0)
    - GoogleUtilities/Environment (~> 6.7)
    - GoogleUtilities/Logger (~> 6.7)
    - nanopb (~> 1.30905.0)
  - FirebaseCrashlytics (4.4.0):
    - FirebaseCore (~> 6.10)
    - FirebaseInstallations (~> 1.6)
    - GoogleDataTransport (~> 7.2)
    - nanopb (~> 1.30905.0)
    - PromisesObjC (~> 1.2)
  - FirebaseInstallations (1.7.0):
    - FirebaseCore (~> 6.10)
    - GoogleUtilities/Environment (~> 6.7)
    - GoogleUtilities/UserDefaults (~> 6.7)
    - PromisesObjC (~> 1.2)
  - FirebaseInstanceID (4.8.0):
    - FirebaseCore (~> 6.10)
    - FirebaseInstallations (~> 1.6)
    - GoogleUtilities/Environment (~> 6.7)
    - GoogleUtilities/UserDefaults (~> 6.7)
  - FirebaseMessaging (4.6.2):
    - FirebaseCore (~> 6.10)
    - FirebaseInstanceID (~> 4.6)
    - GoogleUtilities/AppDelegateSwizzler (~> 6.7)
    - GoogleUtilities/Environment (~> 6.7)
    - GoogleUtilities/Reachability (~> 6.7)
    - GoogleUtilities/UserDefaults (~> 6.7)
    - Protobuf (>= 3.9.2, ~> 3.9)
  - FirebasePerformance (3.3.1):
    - FirebaseCore (~> 6.9)
    - FirebaseInstallations (~> 1.5)
    - FirebaseRemoteConfig (~> 4.7)
    - GoogleDataTransport (~> 7.0)
    - GoogleToolboxForMac/Logger (~> 2.1)
    - "GoogleToolboxForMac/NSData+zlib (~> 2.1)"
    - GoogleUtilities/Environment (~> 6.2)
    - GoogleUtilities/ISASwizzler (~> 6.2)
    - GoogleUtilities/MethodSwizzler (~> 6.2)
    - GTMSessionFetcher/Core (~> 1.1)
    - Protobuf (~> 3.12)
  - FirebaseRemoteConfig (4.9.1):
    - FirebaseABTesting (~> 4.2)
    - FirebaseCore (~> 6.10)
    - FirebaseInstallations (~> 1.6)
    - GoogleUtilities/Environment (~> 6.7)
    - "GoogleUtilities/NSData+zlib (~> 6.7)"
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
 OUTPUT GOES HERE
```

- **Platform that you're experiencing the issue on**:
  - [x] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `e.g. 5.4.3`
- **`Firebase` module(s) you're using that has the issue:**
  - `e.g. Instance ID`
- **Are you using `TypeScript`?**
  - `Y` & `4.0.3`

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by wilau2** [comment]
Testing the following workaround: 
`pod update FirebasePerformance`

**Comment by mikehardy** [comment]
Indeed - I'll be curious for results @wilau2 - for reference https://firebase.google.com/support/release-notes/ios#version_6340_patch_release_-_october_23_2020

**Comment by wilau2** [comment]
So I updated all firebase libs: 

```
-    "@react-native-firebase/app": "8.4.5",
-    "@react-native-firebase/analytics": "7.6.7",
-    "@react-native-firebase/crashlytics": "8.4.9",
-    "@react-native-firebase/messaging": "7.8.11",
-    "@react-native-firebase/perf": "7.4.8",
+    "@react-native-firebase/app": "8.4.6",
+    "@react-native-firebase/analytics": "7.6.8",
+    "@react-native-firebase/crashlytics": "8.4.11",
+    "@react-native-firebase/messaging": "7.9.1",
+    "@react-native-firebase/perf": "7.4.9",
```

I got: 
```
[!] CocoaPods could not find compatible versions for pod "Firebase/CoreOnly":
  In snapshot (Podfile.lock):
    Firebase/CoreOnly (= 6.30.0, ~> 6.30.0)

  In Podfile:
    RNFBApp (from `../node_modules/@react-native-firebase/app`) was resolved to 8.4.6, which depends on
      Firebase/CoreOnly (~> 6.34.0)

```
I ran: `pod update Firebase/CoreOnly`


I ran than ran: `pod update FirebasePerformance`

We halted our iOS production rollout at 2%.

I'll prepare another one I'll keep this thread up to date.

**Comment by wilau2** [comment]
To give an idea why we had to halt: 

![image](https://user-images.githubusercontent.com/5473183/97215485-e176e500-179a-11eb-8cb6-9199da2182d2.png)

**Comment by mikehardy** [comment]
Mmm - you want to remove your Podfile.lock entirely, override sdk version to 6.34.0, pod repo update then pod install to get a full fresh batch rooted in the 6.34.0 version https://rnfirebase.io/#ios

**Comment by jacquesdev** [comment]
@mikehardy - looks like that issue was re-opened saying it's not working yet. So I guess that for right now the best solution is to remove it, and then add it back later when it's confirmed working.

**Comment by mikehardy** [comment]
A shame but thank you very much for tracking this and providing updates, it's a help @jacquesdev

**Comment by wilau2** [comment]
Everything looks much better so far thanks @mikehardy. 
Back to 2% on our new version. 

We decided not to bump our npm packages.

(we have a conflicting protobuff version... 🥱 )

```
# Override Firebase SDK Version
$FirebaseSDKVersion = '6.34.0'
```

So far it looks a lot better than what we had before.

**Comment by Aung-Myint-Thein** [comment]
Hello, I am having similar issue.. this build is working fine till a few days ago and I tried to pod install yesterday and this error come up:

```
[!] CocoaPods could not find compatible versions for pod "Firebase/Performance":
  In Podfile:
    RNFBPerf (from `../node_modules/@react-native-firebase/perf`) was resolved to 7.4.10, which depends on
      Firebase/Performance (~> 6.34.0)

None of your spec sources contain a spec satisfying the dependency: `Firebase/Performance (~> 6.34.0)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
```

I tried adding the following in the podfile..
```
# Override Firebase SDK Version
$FirebaseSDKVersion = '6.34.0'
```

I can see following outputs in the pod install.. hence, I think the above script is working but the above error still showed up and pod install failed.
```
RNFBAnalytics: Using user specified Firebase SDK version '6.34.0'
RNFBApp: Using user specified Firebase SDK version '6.34.0'
RNFBAuth: Using user specified Firebase SDK version '6.34.0'
RNFBCrashlytics: Using user specified Firebase SDK version '6.34.0'
RNFBDynamicLinks: Using user specified Firebase SDK version '6.34.0'
RNFBMessaging: Using user specified Firebase SDK version '6.34.0'
RNFBPerf: Using user specified Firebase SDK version '6.34.0'
```

My rnfirebase lib versions are as following:

"@react-native-firebase/analytics": "^7.6.2",
    "@react-native-firebase/app": "^8.4.2",
    "@react-native-firebase/auth": "^9.2.2",
    "@react-native-firebase/crashlytics": "^8.4.4",
    "@react-native-firebase/dynamic-links": "^7.5.3",
    "@react-native-firebase/messaging": "^7.8.6",
    "@react-native-firebase/perf": "^7.4.9",

However, after I removed the perf, I get the following error:

```
[!] CocoaPods could not find compatible versions for pod "Firebase/Auth":
  In Podfile:
    RNFBAuth (from `../node_modules/@react-native-firebase/auth`) was resolved to 9.3.2, which depends on
      Firebase/Auth (= 6.34.0)

None of your spec sources contain a spec satisfying the dependency: `Firebase/Auth (= 6.34.0)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
```
Then I updated my rnfirebase libraries to 

"@react-native-firebase/analytics": "^7.6.8",
    "@react-native-firebase/app": "^8.4.6",
    "@react-native-firebase/auth": "^9.3.1",
    "@react-native-firebase/crashlytics": "^8.4.11",
    "@react-native-firebase/dynamic-links": "^7.5.10",
    "@react-native-firebase/messaging": "^7.9.1",
    "@react-native-firebase/perf": "^7.4.9",

and still the same pod install error.

Before every retry, I did following commands:

```
rm -rf node_modules && npm install
cd ios
pod deintegrate && pod cache clean --all
rm -rf Podfile.lock
pod install
```

Any recommendation or tips to solve this? Thanks

**Comment by Aung-Myint-Thein** [comment]
I revert back all the changes and set the versions for each rnfirebase libraries as following and now the pod install is successful.

"@react-native-firebase/analytics": "7.6.2",
    "@react-native-firebase/app": "8.4.2",
    "@react-native-firebase/auth": "9.2.2",
    "@react-native-firebase/crashlytics": "8.4.4",
    "@react-native-firebase/dynamic-links": "7.5.3",
    "@react-native-firebase/messaging": "7.8.6",
    "@react-native-firebase/perf": "7.4.3",

**Comment by mikehardy** [comment]
@Aung-Myint-Thein unfortunately, that's the wrong solution. You just reverted back the pod version requested, you did not fix that you do not have the 6.34.0 pod definitions available to you locally, when you should. `pod repo update`, `rm Podfile.lock`, `pod install`. Should resolve it.
